File: typescript/Side/03_difference-of-squares/difference-of-squares.ts
class Squares {

    // Public Properties
    
    public squareOfSum: number;
    public sumOfSquares: number;
    public difference: number;

    // Init

    constructor(side: number) {
        let squareOfSum = this.calculateSquareOfSum(side);
        let sumOfSquares = this.calculateSumOfSquares(side);

        this.squareOfSum = squareOfSum;
        this.sumOfSquares = sumOfSquares;
        this.difference = squareOfSum - sumOfSquares;
    }

    private calculateSquareOfSum(number: number) {
        let returnValue = 0;        
        for (var _i = 0; _i <= number; _i++) {
            returnValue += _i;
        }
        
        return returnValue * returnValue;
    }

    private calculateSumOfSquares(number: number) {
        let returnValue = 0;        
        for (var _i = 0; _i <= number; _i++) {
            returnValue += _i * _i;
        }
        
        return returnValue;
    }
}
  
export default Squares
  

File: swift/Main/05_difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
final class Squares {

    // MARK: - Properties

    private let number: Int

    lazy var squareOfSum: Int = {
        let reducedSum = (0 ... self.number).reduce(0, +)
        return reducedSum * reducedSum
    }()

    lazy var sumOfSquares: Int = {
        return (0 ... self.number).reduce(0, { $0 + $1 * $1 })
    }()

    lazy var differenceOfSquares: Int = {
        return self.squareOfSum - self.sumOfSquares
    }()

    // MARK: - Init

    init(_ side: Int) {
        self.number = side
    }

}
